### ISSUE TITLE
Problems setting Window.Owner when starting application

### ISSUE THREAD

**Issue by bender2k14** [elmish/Elmish.WPF#206]
Issue original reported in https://github.com/elmish/Elmish.WPF/issues/200#issuecomment-618089747.

For the `NewWindow` sample, if the message `ShowWin2` is returned by `init`, then there is a `NullReferenceException` because `Application.Current` is `null` in
https://github.com/elmish/Elmish.WPF/blob/66c909041a09c3d3e64326893459a8cdc1ea9a45/src/Samples/NewWindow/Program.fs#L104

MWE given in the branch https://github.com/mbenoni7/Elmish.WPF/tree/init-cmd-bug-pt-2

**Comment by bender2k14** [comment]
PR #207 makes some progress, but fixing the `NullReferenceException` uncovers another issue.

After making that change, there is a `InvalidOperationException` because [`Window.Owner` ](https://docs.microsoft.com/en-us/dotnet/api/system.windows.window.owner?view=netframework-4.8) is being
> ...set with a window that has not been previously shown.

It is not obvious to me how to avoid this.  I will keep thinking about it.

**Comment by cmeeren** [comment]
Brainwave, this may not make sense (I'm not that into the code at the moment). But won't everything be fine as long as we guarantee that `Application` is instantiated and the main window shown before we start the Elmish loop?

**Comment by bender2k14** [comment]
Yes, that should work.  I will try that.
